packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -85,123 +109,507 @@ const propTypes = {
    * The selected or entered date value to display in the date input.
    */
   value: PropTypes.string,
-  /**
-  * String that labels the current element. 'aria-label' must be present,
-  * for accessibility.
-  */
-  ariaLabel: PropTypes.string,
 };
 
 const defaultProps = {
+  ariaLabel: undefined,
   buttonRefCallback: undefined,
+  id: undefined,
   inputAttributes: undefined,
+  inputRefCallback: undefined,
   isIncomplete: false,
   isInvalid: false,
   name: undefined,
   onBlur: undefined,
+  onButtonFocus: undefined,
   onChange: undefined,
   onClick: undefined,
   onFocus: undefined,
-  onButtonFocus: undefined,
   onKeyDown: undefined,
   required: false,
   useExternalFormatMask: false,
   value: undefined,
-  ariaLabel: undefined,
 };
 
-class DatePickerInput extends React.Component {
-  constructor(props) {
-    super(props);
+const dateReducer = (state, inputValue) => ({
+  ...state,
+  ...inputValue,
+});
+
+const DatePickerInput = (props) => {
+  const {
+    ariaLabel,
+    buttonRefCallback,
+    id,
+    inputAttributes,
+    inputRefCallback,
+    intl,
+    isIncomplete,
+    isInvalid,
+    name,
+    onBlur,
+    onButtonFocus,
+    onChange,
+    onClick,
+    onFocus,
+    onKeyDown,
+    required,
+    useExternalFormatMask,
+    value,
+    ...customProps
+  } = props;
+
+  const [date, dateDispatch] = useReducer(dateReducer, { day: '', month: '', year: '' });
+  const [isFocused, setFocused] = useState(false);
+  const [dayInitialFocused, setDayInitialFocused] = useState(false);
+  const [monthInitialFocused, setMonthInitialFocused] = useState(false);
+  const [yearInitialFocused, setYearInitialFocused] = useState(false);
+  const dayInputRef = useRef();
+  const monthInputRef = useRef();
+  const yearInputRef = useRef();
+  const uuidRef = useRef(uuidv4());
+  const theme = React.useContext(ThemeContext);
+
+  const { onCalendarButtonClick, shouldShowPicker } = customProps;
+  delete customProps.onCalendarButtonClick;
+  delete customProps.shouldShowPicker;
+
+  const additionalInputProps = { ...customProps, ...inputAttributes };
+  const momentDateFormat = DateUtil.getFormatByLocale(intl.locale);
+  const dateValue = DateUtil.convertToISO8601(value, momentDateFormat);
+  const dateFormatOrder = intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' });
+
+  const setDayRef = useCallback(node => {
+    dayInputRef.current = node;
+    if (inputRefCallback && dateFormatOrder === 'YMD') {
```

Comment:
can we move these format order strings to enum or something. instead of hardcoding it everywhere ?

Reply:
Changed here: https://github.com/cerner/terra-framework/pull/1208/commits/12cca2fe9629568cfcd669217a2d553972c230e9